packages/fclasses/src/Design.tsx

```diff
@@ -25,7 +25,7 @@ export type DesignElement<P> = (c: ComponentType<P> | string) => ComponentType<P
  * This is the type to use for the components prop of a component with a fluid design.
  */
 export type DesignableComponents = {
-  [key: string]: ComponentType<any>,
+  [key: string]: ComponentType<any> | null,
```

Comment:
Do we need null?  What is the advantage over Fragment?  Also, if we're going to change this API, can we add 'string'  (or, even better, `keyof JSX.IntrinsicElements` also?  I'm getting tited of adding 'asComponent' every time i want to put an element in (though this might propagate all kinds of type errors across the platform -- if so, we'll put it in a different ticket.

Reply:
the use case for null is when we have a dependent component. For example, StartingTrail and Separator. We want to render Separator only if StartingTrail is set. If set StartingTrail to Fragment, then during rendering phase we can not determine if the component is Fragment or not.
but once I added a null here, I broke type validation in other places. for instance, withDesign usage, so probably I need to revert it here and override types just for breadcrumbs